spec/models/collection_spec.rb

```diff
@@ -177,28 +179,29 @@
       let(:index) { 4 }
 
       it 'inserts successfully' do
-        expect(collection.collection_items.count).to eq 4
-        expect(collection.collection_items.last.item).to eq manifestation
+        expect(inserted_item).to have_attributes(seqno: 4, item: manifestation)
+        expect(collection.collection_items).to eq [first_item, second_item, third_item, inserted_item]
+        expect(collection.collection_items.map(&:seqno)).to eq [1, 2, 3, 4]
       end
     end
 
     context 'when insert at the beginning of the list' do
       let(:index) { 1 }
 
       it 'inserts successfully' do
-        expect(collection.collection_items.count).to eq 4
-        expect(collection.collection_items.first.item).to eq manifestation
+        expect(inserted_item).to have_attributes(seqno: 1, item: manifestation)
+        expect(collection.collection_items).to eq [inserted_item, first_item, second_item, third_item]
+        expect(collection.collection_items.map(&:seqno)).to eq [1, 2, 3, 4]
       end
     end
 
     context 'when insert at specified position' do
       let(:index) { 2 }
 
       it 'inserts successfully' do
-        expect(collection.collection_items.count).to eq 4
-        expect(collection.collection_items.map(&:item)).to eq(
-          [first_item.item, manifestation, third_item.item, second_item.item]
```

Comment:
There were a bug, but it passed unnoticed as all collection_items created by factory had nulls items. Probably it was one of the problems with items reordering mentioned here: https://github.com/abartov/bybeconv/issues/529